frontend/app/work_packages/controllers/menus/query-select-dropdown-menu-controller.js

```diff
@@ -0,0 +1,197 @@
+module.exports = function($scope, $sce, LABEL_MAX_CHARS, KEY_CODES) {
+  var scope = $scope;
+  scope.$watch('groups', refreshFilteredGroups);
+  scope.$watch('selectedId', selectTitle);
+
+  function refreshFilteredGroups() {
+    if(scope.groups){
+      initFilteredModels();
+    }
+  }
+
+  function selectTitle() {
+    angular.forEach(scope.filteredGroups, function(group) {
+      if(group.models.length) {
+        angular.forEach(group.models, function(model){
+          model.highlighted = model.id == scope.selectedId;
+        });
+      }
+    });
+  }
+
+  function initFilteredModels() {
+    scope.filteredGroups = angular.copy(scope.groups);
+    angular.forEach(scope.filteredGroups, function(group) {
+      group.models = group.models.map(function(model){
+        return {
+          label: model[0],
+          labelHtml: $sce.trustAsHtml(truncate(model[0], LABEL_MAX_CHARS)),
+          id: model[1],
+          highlighted: false
+        };
+      });
+    });
+  }
+
+  function labelHtml(label, filterBy) {
+    filterBy = filterBy.toLowerCase();
+    label = truncate(label, LABEL_MAX_CHARS);
+    if(label.toLowerCase().indexOf(filterBy) >= 0) {
+      var labelHtml = label.substr(0, label.toLowerCase().indexOf(filterBy))
+        + "<span class='filter-selection'>" + label.substr(label.toLowerCase().indexOf(filterBy), filterBy.length) + "</span>"
```

Comment:
Line is too long.<br>Mixed double and single quotes.<br>Bad line breaking before '+'.